iceoryx_examples/automotive_soa/include/owl/types.hpp

```diff
@@ -0,0 +1,126 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_TYPES_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_TYPES_HPP
+
+#include "iceoryx_hoofs/cxx/function.hpp"
+#include "iceoryx_hoofs/cxx/optional.hpp"
+#include "iceoryx_hoofs/cxx/string.hpp"
+#include "iceoryx_posh/capro/service_description.hpp"
+#include "iceoryx_posh/popo/sample.hpp"
+
+#include "kom/sample_pointer.hpp"
+
+#include <future>
+#include <memory>
+
+namespace owl
+{
+class Runtime;
+namespace core
+{
+template <typename T>
+using Result = T;
+
+using String = iox::cxx::string<100>;
+} // namespace core
+
+namespace kom
+{
+template <typename T>
+using SamplePtr = iox::cxx::optional<iox::popo::Sample<const T>>;
+
+using ServiceIdentifier = core::String;
+using InstanceIdentifier = core::String;
+
+using EventIdentifier = core::String;
+using FieldIdentifier = core::String;
+using MethodIdentifier = core::String;
+
+struct FindServiceCallbackHandle
+{
+    // Only the Runtime shall be able to create handles, not the user
+    friend class owl::Runtime;
+
+  public:
+    const ServiceIdentifier& GetServiceId() const noexcept
+    {
+        return m_serviceIdentifier;
+    }
+    const InstanceIdentifier& GetInstanceId() const noexcept
+    {
+        return m_instanceIdentifier;
+    }
+
+  private:
+    FindServiceCallbackHandle(ServiceIdentifier serviceIdentifier, InstanceIdentifier instanceIdentifier) noexcept
+        : m_serviceIdentifier(serviceIdentifier)
+        , m_instanceIdentifier(instanceIdentifier)
+    {
+    }
+    ServiceIdentifier m_serviceIdentifier;
+    InstanceIdentifier m_instanceIdentifier;
+};
+
+struct ProxyHandleType
+{
+    // Only the Runtime shall be able to create handles, not the user
+    friend class owl::Runtime;
+
+  public:
+    bool operator==(const ProxyHandleType& rhs) noexcept
+    {
+        return (m_serviceIdentifier == rhs.m_serviceIdentifier) && (m_instanceIdentifier == rhs.m_instanceIdentifier);
+    }
+    const ServiceIdentifier& GetServiceId() const noexcept
+    {
+        return m_serviceIdentifier;
+    }
+    const InstanceIdentifier& GetInstanceId() const noexcept
+    {
+        return m_instanceIdentifier;
+    }
+
+  private:
+    ProxyHandleType(ServiceIdentifier serviceIdentifier, InstanceIdentifier instanceIdentifier) noexcept
+        : m_serviceIdentifier(serviceIdentifier)
+        , m_instanceIdentifier(instanceIdentifier)
+    {
+    }
+    ServiceIdentifier m_serviceIdentifier;
+    InstanceIdentifier m_instanceIdentifier;
+};
+
+template <typename T>
+using ServiceHandleContainer = iox::cxx::vector<T, 50U>;
```

Comment:
What does the 50 means. Is it the maximum supported amount of search results?

Reply:
Magic numbers are magic :rainbow: Wrapped it in a constexpr.